### PR TITLE
Eliminate max size param for Component trait, switch to glam for linear algebra

### PR DESCRIPTION
**Pull request by Masterchef365**
This is a big one. This PR makes it so that Component data types are enforced to be fixed size, thus eliminating the need to include the "maximum size" of a component at compile time. This is important because it makes it much easier for plugin developers to add components. A side effect of this is that we cannot use `nalgebra`'s types in components, because they are serialized as sequences instead of as tuples (they are counted as variable size, even though they may not be). So we switched to the `glam` library for linear algebra instead, which serializes with fixed size. This PR also contains code for verifying that a given type is fixed-size under bincode's rules.

### CODE REVIEW

**Comment by Exiled1** [comment]
Lmao, I don't know what I'm reviewing here, I was there with you making these changes 😂

**Comment by Masterchef365** [comment]
> Lmao, I don't know what I'm reviewing here, I was there with you making these changes joy

I wrote a little more last night! Specifically, I implemented the validation you wrote into the interface and gave it nicer looking error messages

**Comment by Exiled1** [comment]
Oh! Cool then! Will do

File: common/src/lib.rs [hunk 1]
```diff
@@ -25,16 +25,13 @@ pub mod vr;
...
-    pub orient: UnitQuaternion<f32>,
+    pub orient: Quat,
 }
 
 impl Component for Transform {
-    const ID: ComponentIdStatic = ComponentIdStatic {
-        id: pkg_namespace!("Transform"),
-        size: 44,
-    };
+    const ID: &'static str = pkg_namespace!("Transform");
```
Comment (Exiled1) [inline common/src/lib.rs:34]: All hail the changes

File: common/src/lib.rs [hunk 2]
```diff
@@ -146,29 +151,24 @@ macro_rules! make_handle {
...
 
-impl Mul for Transform {
-    type Output = Self;
-    fn mul(self, rhs: Self) -> Self::Output {
-        let lhs: Isometry3<f32> = self.into();
-        let rhs: Isometry3<f32> = rhs.into();
-        (lhs * rhs).into()
+impl Default for GenericHandle {
+    fn default() -> Self {
+        Self(0xBAD_BAD_BAD_BAD_BAD_BAD_BAD_BAD_BAD_BAD)
```
Comment (Exiled1) [inline common/src/lib.rs:172]: Bad bad

Comment (Exiled1) [reply common/src/lib.rs:172]: Also, a user won't have to do this right?

File: engine_interface/src/component_validate_error.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+impl de::Error for ValidationError {
+    // Don't ask... This stuff just exists to make the compiler happy.
+    fn custom<T: Display>(_msg: T) -> Self {
+        panic!("Custom error unsupported")
+    }
+}
+
+impl Display for ValidationError {
```
Comment (Exiled1) [inline engine_interface/src/component_validate_error.rs:30]: Ah, this is what they're for

File: engine_interface/tests/component_validation_test.rs [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+#[test]
+#[should_panic]
+fn ser_none() {
+    let a: Option<()> = None;
+    is_fixed_size(a).unwrap();
+}
+
+#[test]
+#[should_panic]
+fn ser_some() {
```
Comment (Exiled1) [inline engine_interface/tests/component_validation_test.rs:113]: Did you make a check for fixed option?

**Review by Exiled1 — APPROVED** [review]
What a change... All that for a drop() of size. Tbh Totally worth it, now users won't be like, what the fuck why do I have to crash before i can develop?

File: common/src/lib.rs (hunk 1 above)
Comment (Masterchef365) [reply common/src/lib.rs:34]: Amen :pray:

File: common/src/lib.rs (hunk 2 above)
Comment (Masterchef365) [reply common/src/lib.rs:172]: Nah they don't need to, this is just to make sure that if the default GenericHandle is found in e.g. a debugger it's recognizable

File: engine_interface/tests/component_validation_test.rs (hunk 4 above)
Comment (Masterchef365) [reply engine_interface/tests/component_validation_test.rs:113]: I forgor :skull:

**Pull request merged**